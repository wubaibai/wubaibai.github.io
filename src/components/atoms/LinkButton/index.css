.button {
	display: inline-block;
	background: var(--mediumBlue);
	font-family: var(--fontNarrow);
	border: none;
	color: white;
	padding: 8px 20px;
	font-size: 20px;
	font-weight: 500;
	line-height: 1;
	border-radius: 100px;
	text-decoration: none;
	box-shadow: #dedede 0 1px 3px;
	transition: background 0.1s;

	&:hover {
		background: var(--acent);
	}
}
