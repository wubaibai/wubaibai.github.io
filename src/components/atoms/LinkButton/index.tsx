import React from 'react';
import classNames from 'classnames';

import style from './index.css';

interface LinkButtonProperty {
	to?: string;
	children: React.ReactChildren | string;
	className?: string;
}

const LinkButton: React.FC<LinkButtonProperty> = ({ to, children, className }) => {
	return (
		// eslint-disable-next-line react/jsx-no-target-blank
		<a className={classNames(style.button, className)} href={to} role="button" target="_blank">
			{children}
		</a>
	);
};

export default LinkButton;
