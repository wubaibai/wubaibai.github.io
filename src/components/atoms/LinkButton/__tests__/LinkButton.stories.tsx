import React from 'react';
import { storiesOf } from '@storybook/react';

import LinkButton from 'components/atoms/LinkButton';

const stories = storiesOf('atoms/LinkButton', module);

stories.add('__interactive', () => <LinkButton>查看履歷</LinkButton>, { jest: 'LinkButton' });
