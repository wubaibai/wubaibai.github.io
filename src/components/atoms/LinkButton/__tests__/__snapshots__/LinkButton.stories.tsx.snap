// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <a
    class="button"
    role="button"
    target="_blank"
  >
    查看履歷
  </a>
</DocumentFragment>
`;
